### ISSUE TITLE
Files from _static are not copied automatically

### ISSUE THREAD

**Issue by martin-ueding** [sphinx-doc/sphinx#1810]
I have a bunch of files in `_static` and in `_themes/mu/static` that do not get copied when I compile:

```
PYTHONPATH=. /usr/share/sphinx/scripts/python3/sphinx-build -q -b html -d _build/doctrees/en -c _config/en . _build/html/en
```

In my project, I generate a `.css` file from a `.sass` file with a makefile and then call the above. The new `.css` file from the `_themes/mu/static` is not copied. When I do a clean build, everything is copied, but the change detection that Sphinx Doc uses seems to be insufficient.

**Comment by szabolcsdombi** [comment]
Have you found the answer?

**Comment by martin-ueding** [comment]
Not really, I just have a `Makefile` that then copies those files for me. Not really nice, but it seems to work.

**Comment by tk0miya** [comment]
Hi,

It was reported as https://github.com/sphinx-doc/sphinx/issues/1810 .
But, unfortunately it is not fixed yet.
Please send us pull request if you are interested in :-)

Thanks,

Takeshi KOMIYA

2017-02-12 0:13 GMT+09:00 Martin Ueding <notifications@github.com>:

**Comment by szabolcsdombi** [comment]
added 2 lines to the `conf.py`

```
def setup(app):
    app.add_stylesheet('css/custom.css')
```

**copy works fine with the css files**

*source/_static/css/custom.css* is copied to the *build/html/_static/css/custom.css*

**Comment by tk0miya** [comment]
`add_stylesheet()` has same problem: #2090.

**Comment by ghost** [comment]
any resolution on this?

**Comment by im-n1** [comment]
Just ran into this. I consider this quite important. Here is my real world example.

Imagine you document an UI kit. For every component you document you create and example HTML which you include with iframe (`.. raw::` directive) because  you need a separated DOM for custom  styles and JS. Then you place this HTML into a file under `_static` folder and you refer to it like `<iframe src="_static/example.html"></iframe>`. Now it's part of your documentation just like any other rst file.

Static files needs to be copied automatically (or at least a new CLI swith to turn this functionality on) in certain cases.

**Comment by basnijholt** [comment]
@im-n1, I ran into this issue too in https://github.com/python-adaptive/adaptive/pull/329.

Did you find a workaround?

**Comment by pradyunsg** [comment]
From https://github.com/executablebooks/sphinx-autobuild/#relevant-sphinx-bugs:

> At the time of writing, the only known workaround is to instruct Sphinx to rebuild the relevant pages. This can be done by disabling incremental mode (with `-a`) or passing relevant filenames in addition to source and output directory in the CLI.

**Comment by pradyunsg** [comment]
@AA-Turner what change in Sphinx fixed this? I wanted to add a reference when closing the sphinx-autobuild issue, but couldn't figure out what the relevant issue is.

**Comment by AA-Turner** [comment]
This didn't have a reproducer, but a cursory investigation indicates that ``copy_html_static_files`` should copy all of the files (except those with a leading ``.``) from ``html_static_path`` to ``outdir/_static``.

There may be something I'm missing?

A

**Comment by pradyunsg** [comment]
This isn't static files from html_static_path, but rather from the theme.

An easy reproducer would be to:
- build a Sphinx documentation as html
- modify the HTML theme to have a red background in body
- Build again
- The rebuilt site doesn't have any rendering changes
- Add a new page
- Build again
- The site's new page will be rendered with the new red background.

Sphinx has an assumption that the theme wouldn't be modified, which means what modifying the theme isn't reflected in the build results.

**Comment by pradyunsg** [comment]
FWIW, this is relatively easy to hit during theme development with sphinx-autobuild which... as you'd expect, involves modifying theme files a lot. :)

**Comment by pradyunsg** [comment]
Adding a cross-reference to https://github.com/sphinx-doc/sphinx/issues/2090, which is somewhat related (albeit for stylesheets).

**Comment by Sieboldianus** [comment]
> FWIW, this is relatively easy to hit during theme development with sphinx-autobuild which... as you'd expect, involves modifying theme files a lot. :)

Like me.. the solution currently is to delete the `_build` folder entirely after updating `custom.css` (e.g.) and then rebuilding manually.
